services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -203,6 +205,10 @@ async def list_files(  # noqa C901
                     for node_id, node_data in proj_data.workbench.items()
                 }
 
+        # expanding directories
+        for entry in delayed_expands:
```

Comment:
1. Yes entry is a coroutine.

2. The issue is that `expand_directory` runs using S3. While this is running the connection to the database is opened.

Since it is not required to keep the connection/transaction open while listing the files in S3, this simply avoids doing so.